cmd/shfmt/main_test.go

```diff
@@ -60,13 +61,13 @@ func TestStdin(t *testing.T) {
 		if err := formatStdin(); err != errChangedWithDiff {
 			t.Fatalf("got=%q want=%q", err, errChangedWithDiff)
 		}
-		want := `diff -u <standard input>.orig <standard input>
+		want := fmt.Sprintf(`diff -u <standard input>.orig <standard input>
 @@ -1,3 +1,2 @@
-- foo
-+foo
+%s
+%s
  bar
--
-`
+%s
+`, markDeleted([]byte("- foo")), markAdded([]byte("+foo")), markDeleted([]byte("-")))
```

Comment:
Now that we'll have diff with and without colors, you could make `color` a global bool like `diff`, and have two subtests:

`Diff`, which is the old test as-is
`DiffColored`, which sets `color = true`, and ensures the escape codes are present.

Also, don't use `fmt.Sprintf` for the expected output. Just write the entire string as a single literal, escape codes and all.